### ISSUE TITLE
PID file gets deleted after a reload

### ISSUE THREAD

**Issue by masterkorp** [Unitech/PM2#1285]
The PID file of a process gets deleted after a ``pm2 reload``, this can become very annoying if you run checks to see if the application is running based on the PID file.

**How to reproduce:**
````
pm2 describe app
cat /path/to/pidfile.pid
pm2 reload app
cat /path/to/pidfile.pid
cat: /path/to/pidfile.pid: No such file or directory
```

**pm2 version:** ``0.12.15``
**node version:** ``v1.7.1`` (io.js)
**OS:** Ubuntu 14.04

**Comment by jshkurti** [comment]
Indeed, gonna fix it.
I'll keep you informed.
Thanks for the feedback

**Comment by masterkorp** [comment]
Hello @jshkurti , any news on this?

I won't mind providing a patch on this. If someone pointed me in the right direction?
Where shall I start looking ?
What is the recommended way of setting up a development environment ?

**Comment by jshkurti** [comment]
Here we go :)

**Comment by jshkurti** [comment]
`npm i -g git://github.com/Unitech/PM2#development` for now, it will be included on v0.12.16 otherwise :)
